.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,130 @@
+name: deploy
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  terraform-deploy:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: write
+      contents: read
+      pull-requests: write
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Retrieve secret from Vault
+        uses: hashicorp/vault-action@v2.5.0
+        with:
+          method: jwt
+          url: "https://quansight-public-vault-fe415d04.c219cf75.z1.hashicorp.cloud:8200"
```

Comment:
Shall we have this as a repo/deployment secret instead?

Reply:
Honestly none of this information needs to be secret but it would be nice if the url and namespace were organization secrets to make the github-actions more reusable if we were to change the url.